revisions-cli.php

```diff
@@ -76,10 +80,62 @@ public function dump( $args = array(), $assoc_args = array() ) {
 	 */
 	public function list_( $args = array(), $assoc_args = array() ) {
 
+		// Default fields to return.
+		$fields = [
+			'ID',
+			'post_title',
+			'post_parent',
+		];
+
+		// Customise the fields that are returned.
+		if ( ! empty( $assoc_args['fields'] ) ) {
+
+			$allowed_fields = [
+				'ID',
+				'post_author',
+				'post_date',
+				'post_date_gmt',
+				'post_content',
+				'post_title',
+				'post_excerpt',
+				'post_status',
+				'comment_status',
+				'ping_status',
+				'post_name'.
+				'post_modified',
+				'post_modified_gmt',
+				'post_password',
+				'to_ping',
+				'pinged',
+				'post_content_filtered',
+				'post_parent',
+				'guid',
+				'menu_order',
+				'post_type',
+				'post_mime_type',
+				'comment_count',
+			];
+
+			// Don't allow fields that aren't in the above whitelist.
+			$excluded_fields = array_diff( $assoc_args['fields'], $allowed_fields );
+			if ( ! empty( $excluded_fields ) ) {
+				WP_CLI::error( 'Invalid values provided in the fields argument.' );
+			}
+
+			$fields = implode( ',', $assoc_args['fields'] );
+
+		}
+
 		global $wpdb;
 		if ( ! empty( $assoc_args['post_id'] ) ) {
 
-			$revs = $wpdb->get_results( $wpdb->prepare( "SELECT ID, post_title, post_parent FROM $wpdb->posts WHERE post_parent = %d", $assoc_args['post_id'] ) );
+			$revs = $wpdb->get_results(
+				$wpdb->prepare(
+					"SELECT %s FROM $wpdb->posts WHERE post_parent = %d",
+					$fields,
```

Comment:
`$fields` can't be sprintf'd into the SQL... as a string it gets quoted, as an array it throws an error.